### DISCUSSION TITLE
wsub in aerosol activation

### DISCUSSION THREAD

**Discussion by JulioTBacmeister** [NCAR/amwg_dev#348] [TF2: Tuning]
@adamrher @tilmes @cecilehannay @PeterHjortLauritzen 
Vince Larson and Ben Stephens pointed out an interesting feature of the aerosol activation routine dropmixnuc in ndrop.F90.  It actually does another operation on wsub which is fed in from microp_aero.F90. Lines 560-564 in ndrop.F90:

         wtke_cen(i,k) = wsub(i,k)
         wtke(i,k)     = wsub(i,k)
         wtke_cen(i,k) = max(wtke_cen(i,k), wmixmin)
         wtke(i,k)     = max(wtke(i,k), wmixmin)

The floor value 'wmixmin' is hardwired at 0.1 m/s (line 338).  

My new understanding of this code then is that the activation NEVER sees a subgrid vertical of less than 0.1 m/s, regardless of what what is done with the namelist parameters we've been fiddling with. Terrible coding, but good news in that the big impacts we saw in #330 are not the result of having WSUB~0.0 everywhere.

**Comment by adamrher** [comment]
To restore some sanity we should probably make sure ```microp_aero_wsub_min```=wmixmin so the two limiters are consistent.

**Comment by JulioTBacmeister** [comment]
We can discuss this.  I agree it is dumb to have a floor value that is then ignored by a different floor value.

**Comment by bstephens82** [comment]
I tried a simple test where I [hard-coded wmixmin=0.01](https://github.com/ESCOMP/CAM/blob/e0a047882db4b2ce2d721494a26a34ff249ee3ba/src/physics/cam/ndrop.F90#L226), and I also used the namelist setting wsub_scale=0.01.  This drastically lowered the magnitude of SWCF:
![unnamed](https://github.com/NCAR/amwg_dev/assets/59519056/cf928d9f-1512-41df-995c-c3faf4f9c369)

Note that the "control" run here also had wsub_scale=0.01.  So the only change between the two runs was the value of wmixmin in ndrop.F90.

**Reply by adamrher** [reply]
That makes sense to me. If you had not changed wsub_scale=0.01 in the experiments, it wouldn't be so sensitive to the hard coded floor value that you've changed.

**Reply by adamrher** [reply]
It kind of makes sense to me to multiply the hard coded wminxmin=0.1 by the wsub_scale factor. That way, lowering wsub_scale will also lower the floor value so that we aren't constantly hitting the 0.1 floor value.

**Reply by bstephens82** [reply]
Ah I see.  Yeah this was just me pushing buttons to find out what they do haha.
